### PR TITLE
Enable archive item deletion through the API

### PR DESCRIPTION
**Pull request by amickan**
Part of https://github.com/DIAGNijmegen/rse-roadmap/issues/274

After merging this, permissions for existing archive items need to be updated by running `assign_permissions()` on all items.

### CODE REVIEW

**Comment by jmsmkn** [comment]
Why does this need to be an API endpoint? I would have thought that the normal view would be sufficient. That way, you can get confirmation of the consequences, I don't really see the need to automate this from another application.

File: app/tests/archives_tests/test_views.py [hunk 1]
```diff
@@ -849,3 +849,25 @@ def test_archive_item_add_value(
             )
     assert response.status_code == 200
     assert ArchiveItem.objects.get().values.first().value
+
+
+@pytest.mark.django_db
+def test_api_archive_item_deletion(client):
```
Comment (jmsmkn) [inline app/tests/archives_tests/test_views.py:855]: Needs tests for object based permissions, e.g.
- ensuring that a user can only delete an archive item if they have the "delete_archite_item" bit 
- ensuring that the only users and groups with the "delete_archite_item" bit are the editors group for this reader study
See usages of `get_groups_with_set_perms`

**Comment by amickan** [comment]
> Why does this need to be an API endpoint?

I went with the API view since we discussed wanting a bulk delete option in the future (https://github.com/comic/grand-challenge.org/issues/3169). And this is also how display set deleting is done at the moment (although that's not a reason to use an API view per se of course).

**Comment by jmsmkn** [comment]
A bulk delete should be a single view, you shouldn't implement bulk deletion by iterating over individual deletion endpoints, that would take forever.

**Comment by jmsmkn** [comment]
> The latter is not a reason to use an API view per se of course, but if we go for a normal delete view for archive items, we should do that for display sets as well, or else it'll be difficult to use the same templates etc. across the apps

I would rather use a normal view for both. API views are only for things that need to go into other apps (e.g. CIRRUS, user upload scripts via GC API).

**Comment by amickan** [comment]
> I would rather use a normal view for both.

Alright. I understand. I will add normal delete views for both then.

**Comment by amickan** [comment]
I will create a new branch for the delete views.

**Pull request closed without merging**